src/UI/FileTreeView.re

```diff
@@ -75,32 +82,59 @@ let nodeView = (~font: Core.UiFont.t, ~fg, ~bg, ~node: FsTreeNode.t, ()) => {
       <FontIcon
         color=fg
         icon={isOpen ? FontAwesome.folderOpen : FontAwesome.folder}
-        backgroundColor=bg
+        backgroundColor=Colors.transparentWhite
       />
     | _ => <icon />
     };
 
   <View style=Styles.item>
     <icon />
-    <Text text={node.displayName} style={Styles.text(~fg, ~bg, ~font)} />
+    <Text text={node.displayName} style={Styles.text(~fg, ~font)} />
   </View>;
 };
 
 module TreeView = TreeView.Make(FsTreeNode.Model);
 
-let make = (~title, ~tree: FsTreeNode.t, ~onNodeClick, ~state: State.t, ()) => {
+let make =
+    (
+      ~title,
+      ~tree: FsTreeNode.t,
+      ~focus: option(string),
+      ~onNodeClick,
+      ~state: State.t,
+      (),
+    ) => {
   [@warning "-27"]
   let State.{theme, uiFont as font, _} = state;
 
   let bg = state.theme.sideBarBackground;
   let fg = state.theme.sideBarForeground;
 
+  let focus =
+    switch (focus, state.workspace) {
+    | (Some(path), Some(workspace)) => toNodePath(workspace, tree, path)
+    | _ => None
+    };
```

Comment:
Is this expensive to run every time we render render?

Reply:
Depends how many nodes need to be traversed. In practice I don't think it'll be a problem because the file explorer will likely become impractical to use before there's a noticeable performance impact. For example, expanding `node_modules` isn't going to have an impact since its contents will be skipped if it's not in the path. You have to select a file deep in the hierarchy for it to be traversed, and then it'll still only traverse the nodes that are before it, expanded and whose parent is in the path.

There are still way we can improve it though. The simplest would be to store a hash of the name and compare those instead. We should probably do that anyway since using `displayName` is based on an assumption that might not hold. We could also precompute it, but that's an invariant that might be tricky to maintain. It goes against "single source of truth" so we have to remember to recompute it every time the `foxus` OR `tree` changes.

Reply:
Thanks for the details @glennsl ! Do we have a way to do 'perf' logging today with our new framework? It might be worth wrapping this in a 'perf' block so we can track it. 

> We could also precompute it, but that's an invariant that might be tricky to maintain. It goes against "single source of truth" so we have to remember to recompute it every time the foxus OR tree changes.

Right, I've been thinking about this in the context of a few other places (like the `EditorSurface`) - sometimes there are things we'd really like to cache, but it creates new sources of truth. I wonder if we could have some sort of `useMemo` hook that takes an object, a comparison function, and an 'expensive' calculation, and caches the result of that calculation (or recomputes) based on the comparison result.

Reply:
No action needed for this PR - just thinking out loud)

Reply:
`Log.perf` is still there. I'll check it out and probably add a hashcode to `FsTreeNode` for comparison.

> I wonder if we could have some sort of useMemo hook that takes an object, a comparison function, and an 'expensive' calculation, and caches the result of that calculation (or recomputes) based on the comparison result.

That's a good idea!. `Memo` seems a bit misleading, but it sounds basically like `Lazy.t` with an invalidation condition wrapped up in a `ref` hook.